### PR TITLE
Added Parcel sample and fixed esm output issue

### PR DESCRIPTION
**Pull request by maxburs**
* Added Parcel.js sample
* Fixed kusto language server import order issue in esm output
  * Imports are now for the same files as amd version, and in the same order
* Updated README.md instructions to reflect latest changes
* Turned off yarn "nmHoistingLimits" feature. New build scripts use `require.resolve()` to find dependencies so we don't need this anymore, and it causes issues when using `yarn link` to run samples because we'll have multiple copies of `monaco-kusto` in the tree
* Fixed worker consuming monaco global api
  * package/src/kustoWorker.ts
  * I'm not actually sure if this code was not working, if this means we're importing an extra copy of monaco at runtime, or something else
  * Change was made while I was looking into updating us to not use the global api. Didn't do this because the patch would have been even bigger, and I was running into typing issues.
* Added patch to monaco-editor to fix Parcel compatibility issue
  * PR to upstream this patch here: https://github.com/microsoft/vscode/pull/178179
* Release pipeline now copies README.md into the package folder
* Updated package json with esm entry points, and removed "main" entry point because we don't have a common.js entry point.

### CODE REVIEW

**Review by gil-koifman — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -16,6 +16,10 @@
     },
     "packageManager": "yarn@3.4.1",
     "devDependencies": {
-        "prettier": "^2.8.4"
+        "prettier": "^2.8.4",
+        "process": "^0.11.10"
+    },
+    "resolutions": {
+        "monaco-editor@~0.35.0": "patch:monaco-editor@npm%3A0.35.0#./.yarn/patches/monaco-editor-npm-0.35.0-f4a2faf37d.patch"
```
[the remark this answers is not in the document]
Comment (maxburs) [reply package.json:23]: This is due to a poor interaction between Parcel's bundling logic and a file in monaco-kusto. It's not totally clear to me which one is more at fault.

If you open the patch file I've left comments.

I submitted a patch to vscode that got merged today, so should be able to remove this in the future.

File: samples/parcel/package.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    "scripts": {
+        "start": "parcel index.html",
+        "build": "parcel build index.html"
+    },
+    "alias": {
+        "./MonacoEnvironment.ts": "MonacoEnvironment"
+    },
+    "dependencies": {
+        "@kusto/monaco-kusto": "6.1.1",
+        "monaco-editor": "~0.35.0",
```
[the remark this answers is not in the document]
Comment (maxburs) [reply samples/parcel/package.json:15]: Unfortunately, the resolutions config is global, can't move it here. My PR to vscode was merged so we should be able to remove it in the future.

PR: https://github.com/microsoft/vscode/pull/178179

I've also documented it in the README

File: samples/parcel/index.html [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        <meta charset="UTF-8" />
+        <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+        <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+        <title>Monaco Kusto</title>
+    </head>
+
+    <body>
+        <div id="root"></div>
+        <script type="module">
+            import kustoWorkerUrl from 'url:@kusto/monaco-kusto/release/esm/kusto.worker';
```
[the remark this answers is not in the document]
Comment (maxburs) [reply samples/parcel/index.html:13]: Added a comment. Is it enough?

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (maxburs) [reply package.json:23]: PR: https://github.com/microsoft/vscode/pull/178179

File: samples/parcel/index.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (noam-shapira) [reply samples/parcel/index.html:21]: yep

**Pull request merged**